File: jcv-restassured-example/src/main/java/com/ekino/oss/jcv/example/jcvrestassuredexample/dto/OrderDto.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.example.jcvrestassuredexample.dto;

import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.Instant;
import java.time.ZonedDateTime;
import java.util.List;
import java.util.UUID;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@Data
@EqualsAndHashCode
@FieldDefaults(level = AccessLevel.PRIVATE)
public class OrderDto {

    UUID id;
    String reference;
    Instant createdDate;
    ZonedDateTime shippingDate;
    List<OrderItemDto> items;
}

File: jcv-restassured-kotlin-example/src/main/kotlin/com/ekino/oss/jcv/example/jcvrestassuredexample/dto/OrderDto.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.example.jcvrestassuredexample.dto

import java.time.Instant
import java.time.ZonedDateTime
import java.util.UUID

data class OrderDto(
    var id: UUID? = null,
    var reference: String? = null,
    var createdDate: Instant? = null,
    var shippingDate: ZonedDateTime? = null,
    var items: List<OrderItemDto> = emptyList()
)
